src/components/modal/ModalComposite.tsx

```diff
@@ -0,0 +1,106 @@
+import * as React from 'react';
+import * as _ from 'underscore';
+import { IModalProps, Modal } from './Modal';
+import { IModalHeaderProps, ModalHeader } from './ModalHeader';
+import { IModalFooterProps, ModalFooter } from './ModalFooter';
+import { IModalBackdropProps, ModalBackdrop } from './ModalBackdrop';
+import { IReduxStatePossibleProps } from '../../utils/ReduxUtils';
+import { ModalBackdropConnected } from './ModalBackdropConnected';
+import { ModalHeaderConnected } from './ModalHeaderConnected';
+import { JSXRenderable } from '../../utils/JSXUtils';
+import { ModalConnected } from './ModalConnected';
+import { ModalBody } from './ModalBody';
+
+export interface IModalCompositeOwnProps extends IModalProps, IModalHeaderProps, IModalFooterProps, IModalBackdropProps {
+  modalHeaderChildren?: JSXRenderable;
+  modalHeaderClasses?: string[];
+  modalBodyChildren?: JSXRenderable;
+  modalBodyClasses?: string[];
+  modalFooterChildren?: JSXRenderable;
+  modalFooterClasses?: string[];
+}
+
+export interface IModalCompositeStateProps extends IReduxStatePossibleProps { }
+
+export interface IModalCompositeDispatchProps { }
+
+export interface IModalCompositeProps extends IModalCompositeOwnProps, IModalCompositeStateProps, IModalCompositeDispatchProps { }
+
+export class ModalComposite extends React.Component<IModalCompositeProps, {}> {
+  static defaultProps: Partial<IModalCompositeProps> = {
+    id: _.uniqueId('modal'),
+  };
+
+  render() {
+    return (
+      <div>
+        {this.getModal([this.getModalHeader(), this.getModalBody(), this.getModalFooter()])}
+        {this.getModalBackdrop()}
+      </div>
+    );
+  }
+
+  private getModal(children: JSXRenderable) {
+    const basicProps: IModalProps = {
+      id: this.props.id,
+      classes: this.props.classes,
+    };
+    const onRenderProp = this.props.onRender ? () => this.props.onRender() : undefined;
+    const onDestroyProp = this.props.onDestroy ? () => this.props.onDestroy() : undefined;
+
+    return this.props.withReduxState
+      ? <ModalConnected {...basicProps}>{children}</ModalConnected>
+      : <Modal
+        {...basicProps}
+        isOpened={this.props.isOpened}
+        onRender={onRenderProp}
+        onDestroy={onDestroyProp}
+      >
+        {children}
+      </Modal>;
+  }
+
+  private getModalHeader() {
+    const basicProps: IModalHeaderProps = {
+      id: this.props.id,
+      title: this.props.title,
+      classes: this.props.modalHeaderClasses,
```

Comment:
would it be possible to add a tooltip icon besides the title as of now? If not, could you add it?

Reply:
I think it is something you send to the ModalHeader component by using its children, but I'll look into it to make sure.

Reply:
Ok cool thanks, I think even if it's at the children level, I wonder if we could not have a prop `documentation link` and some tooltip props which would generate the doc icon  the proprer way...otherwise this is something we'll have to do over and over manually i think

Reply:
if you don't need it for your use case though fine by me we can do it when the time comes

Reply:
That is exactly the case actually. Just like we can see in the simple ModalExamples. I have also added an example in the ModalCompositeExamples just to make sure.